sklearn/feature_selection/_from_model.py

```diff
@@ -190,6 +206,16 @@ class SelectFromModel(MetaEstimatorMixin, SelectorMixin, BaseEstimator):
            [-0.02],
            [-0.48],
            [ 1.48]])
+
+    Using a callable to create a selector that can use no more than half
+    of the input features.
+
+    >>> half_callable = lambda X: round(len(X[0]) / 2)
+    >>> half_selector = SelectFromModel(estimator=LogisticRegression(),
+    ...                                 max_features=half_callable)
+    >>> _ = half_selector.fit(X, y)
+    >>> half_selector.max_features_
```

Comment:
I think it's safer to use a function instead of a lambda here, so that the estimator can be pickled by joblib, otherwise it will fail and one needs to use cloudpickle to pickle the estimator.